tutorials/continuous-deployment-pipeline-google-container-engine-with-codeship/index.md

```diff
@@ -0,0 +1,275 @@
+regulatory constraints.
+
+## Before you begin
+
+Take the following steps to enable the Google Container Engine API:
+
+1.  Visit the [Container Engine](https://console.cloud.google.com/projectselector/kubernetes)
+    page in the Google Cloud Platform Console.
+1.  Create or select a project in the [Google Cloud Platform
+    Console](https://console.cloud.google.com/).
+1.  Wait for the API and related services to be enabled, which can take several
+    minutes.
+1.  [Enable billing][billing] for your project.
+
+[billing]: https://support.google.com/cloud/answer/6293499#enable-billing
+
+Make sure you have the following:
+
+1.  [Docker CE Version 17.03.01](https://www.docker.com/community-edition)
+1.  [Codeship Account](https://app.codeship.com/registrations/new)
+1.  [Codeship Jet CLI](https://documentation.codeship.com/pro/builds-and-configuration/cli/)
+1.  [`hello-express` source code](https://github.com/kellyjandrews/hello-express)
+1.  [GitHub](https://github.com/), [Bitbucket](https://bitbucket.org/), [GitLab](https://about.gitlab.com/) account
+
+## Setting up the continuous deployment pipeline
+
+### Step 1: Create service account
+
+The interactions with the Google Cloud API from Codeship require a service
+account with permissions to the Storage and Container services.
+
+Follow these steps to create a service account:
+
+1.  Visit the [Service accounts](https://console.cloud.google.com/projectselector/iam-admin/serviceaccounts)
+    page in the Google Cloud Platform Console.
+1.  Select your project.
+1.  Click `Create Service Account`
+1.  Enter a name
+1.  Click `Select a Role` and choose the following permissions:
+
+    * Project -> Service Account Actor
+    * Container -> Container Engine Admin
+    * Storage -> Storage Admin
+1.  Select `Furnish a new private key` and leave the option on `JSON`.
+1.  Click Create, and then close the dialog once it is created.
+
+After the service account is created, a `JSON` file with your credentials
+automatically downloads to your computer. This file will be used in the next
+step.
+
+### Step 2: Setup environment variables
+
+In the `hello-express` source code is the file `example.env`.  Rename this file
+to `.env`.
+
+Inside the file, you will replace the `...` with your Google Cloud Platform
+project details as follows:
+
+1.  `DEFAULT_ZONE`: Select a default [compute zone](https://cloud.google.com/compute/docs/regions-zones/regions-zones#available)
+    like `us-central1-b`
+1.  `APP_NAME`: Give your application a name, in this case use `hello-express`
+1.  `CONTAINER_CLUSTER`: Give your container cluster a name, in this case use
+    `hello-express-cluster`
+1.  `GOOGLE_PROJECT_ID`: The `Project ID` is found in the [Google Cloud Engine dashboard](https://console.cloud.google.com/home/dashboard)
+1.  `GOOGLE_AUTH_EMAIL`: Use the `Service Account ID` for the [service account](https://console.cloud.google.com/iam-admin/serviceaccounts)
```

Comment:
just curious why do you need this value? you don't need to know service account ID to use it.

Reply:
We currently use it here in our Docker config generator for our push steps - https://github.com/codeship-library/google-cloud-utilities/blob/master/dockercfg-generator/gcr_docker_creds.sh#L12

Let me know if this isn't required here, and I can talk to the owner of that module and make sure it gets updated.

Reply:
Any additional feedback regarding this script?